src/wallet/wallet.cpp

```diff
@@ -1051,14 +1051,13 @@ bool CWalletTx::WriteToDisk(CWalletDB *pwalletdb)
     return pwalletdb->WriteTx(GetHash(), *this);
 }
 
-bool CWallet::WitnessBucketCommitment(uint256 &commitment,
-                                      libzcash::MerklePath &path,
+void CWallet::WitnessBucketCommitment(std::vector<boost::optional<uint256>> commitments,
+                                      std::vector<boost::optional<ZCIncrementalWitness>>& witnesses,
                                       uint256 &final_anchor)
```

Comment:
Why is the first parameter a vector of optionals? What situation would you provide a vector with some elements not present?